pkg/cmd/addon/upgrade.go

```diff
@@ -0,0 +1,160 @@
+	ktypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	extensionsv1alpha1 "github.com/apecloud/kubeblocks/apis/extensions/v1alpha1"
+	"github.com/apecloud/kubeblocks/pkg/constant"
+
+	"github.com/apecloud/kbcli/pkg/cluster"
+	"github.com/apecloud/kbcli/pkg/types"
+	"github.com/apecloud/kbcli/pkg/util"
+)
+
+var addonUpgradeExample = templates.Examples(`
+	# upgrade an addon from default index to latest version
+	kbcli addon upgrade apecloud-mysql 
+
+	# upgrade an addon from default index to latest version and skip KubeBlocks version compatibility check
+	kbcli addon upgrade apecloud-mysql --force
+
+	# upgrade an addon to latest version from a specified index
+	kbcli addon upgrade apecloud-mysql --index my-index
+
+	# upgrade an addon with a specified version default index 
+	kbcli addon upgrade apecloud-mysql --version 0.7.0
+`)
+
+// upgradeOption storage the info to upgrade an addon
+type upgradeOption struct {
+	*installOption
+
+	// currentVersion is the addon current version in KubeBlocks
+	currentVersion string
+	// if independent is true will retain the existing addon and reinstall the new version of the addon.
+	// otherwise the upgrade will be in-place
+	independent bool
+	// prefix is the name prefix to identify the same addon with different version when independent is true
+	prefix string
+}
+
+func newUpgradeOption(f cmdutil.Factory, streams genericiooptions.IOStreams) *upgradeOption {
+	return &upgradeOption{
+		installOption:  newInstallOption(f, streams),
+		currentVersion: "",
+		independent:    false,
+		prefix:         "",
+	}
+}
+
+func newUpgradeCmd(f cmdutil.Factory, streams genericiooptions.IOStreams) *cobra.Command {
+	o := newUpgradeOption(f, streams)
+	cmd := &cobra.Command{
+		Use:               "upgrade",
+		Short:             "Upgrade an existed addon to latest version or a specified version",
+		Args:              cobra.ExactArgs(1),
+		Example:           addonUpgradeExample,
+		ValidArgsFunction: util.ResourceNameCompletionFunc(f, o.GVR),
+		Run: func(cmd *cobra.Command, args []string) {
+			o.name = args[0]
+			util.CheckErr(o.Complete())
+			util.CheckErr(o.Validate())
+			util.CheckErr(o.Run())
+		},
+	}
+	cmd.Flags().BoolVar(&o.force, "force", false, "force upgrade the addon and ignore the version check")
+	cmd.Flags().StringVar(&o.version, "version", "", "specify the addon version")
+	cmd.Flags().StringVar(&o.index, "index", types.DefaultIndexName, "specify the addon index index, use 'kubeblocks' by default")
+	cmd.Flags().BoolVar(&o.independent, "independent", false, "when independent is true, it will retain the existing addon and reinstall the new version of the addon, otherwise the upgrade will be in-place")
+	cmd.Flags().StringVar(&o.prefix, "prefix", "", "prefix is the name prefix to identify the same addon with different version when independent is true")
```

Comment:
I prefer to use `name` to let user specify the new addon name, if not specified and in-place is false, it will use `addonname-version`, and this name will be used as the `resourceNamePrefix` of helm chart.